war/pom.xml

```diff
@@ -96,6 +96,16 @@ THE SOFTWARE.
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.main</groupId>
+      <artifactId>websocket-jetty10</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```

Comment:
This seems to make all of the `JenkinsRule`-based tests start failing with:

```
[ERROR] jenkins.slaves.UnsupportedRemotingAgentEscapeHatchTest.shouldBeAbleToConnectAgentWithUnsupportedVersionWithEscapeHatch  Time elapsed: 0.06 s  <<< ERROR!
java.lang.NoSuchFieldError: CACHE
        at org.eclipse.jetty.http2.hpack.HpackContext.<clinit>(HpackContext.java:126)
        at org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder.getEncodedField(HpackFieldPreEncoder.java:68)
        at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:102)
        at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:108)
        at org.eclipse.jetty.http2.hpack.HpackEncoder.<clinit>(HpackEncoder.java:76)
        at org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder.getEncodedField(HpackFieldPreEncoder.java:38)
        at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:102)
        at org.eclipse.jetty.http.PreEncodedHttpField.<init>(PreEncodedHttpField.java:108)
        at org.eclipse.jetty.http.MimeTypes$Type.<init>(MimeTypes.java:98)
        at org.eclipse.jetty.http.MimeTypes$Type.<clinit>(MimeTypes.java:56)
        at org.eclipse.jetty.http.MimeTypes.<clinit>(MimeTypes.java:175)
        at org.jvnet.hudson.test.JenkinsRule.<clinit>(JenkinsRule.java:2967)
        at jenkins.slaves.UnsupportedRemotingAgentEscapeHatchTest.<init>(UnsupportedRemotingAgentEscapeHatchTest.java:32)
        […]
```

I got the problem to go away with:

```xml
diff --git a/test/pom.xml b/test/pom.xml
index e10e9ea835..1e69250152 100644
--- a/test/pom.xml
+++ b/test/pom.xml
@@ -117,6 +117,12 @@ THE SOFTWARE.
       <version>${project.version}</version>
       <type>executable-war</type>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jenkins-ci.main</groupId>
+          <artifactId>websocket-jetty10</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
     <dependency>
       <groupId>io.jenkins.plugins</groupId>
```

With that fix the `JenkinsRule` and `RealJenkinsRule` tests both seem to work. I think it makes sense to put that in until the test harness moves to Jetty 10.

Reply:
Not sure that is quite right. IIUC now the `test` module is loading Jetty from `websocket-jetty9` rather than `jenkins-test-harness`? We want `websocket-jetty9` to be in the test classpath (actually `websocket-jetty10` itself should be fine as well); we just do not want its transitive dependencies to also be in the test classpath. Maybe they should be using `optional` or `provided` on their deps.

Reply:
Are you going to try that out or are you asking me to?

Reply:
Ah I see you just pushed some commits. Yeah that does seem to fix it for me. Here's hoping for a green PR build. :crossed_fingers: